docker/Dockerfile

```diff
@@ -58,6 +60,34 @@ RUN apt-get update && \
         cpanminus \
         iproute2
 
+# Go and go-pluginserver
+ENV GO_VERSION=${GO_VERSION}
+ENV GOROOT=${BUILD_PREFIX}/go
+ENV GOPATH=${BUILD_PREFIX}/gopath
+ENV PATH=$GOPATH/bin:${GOROOT}/bin:$PATH
+RUN mkdir -p ${GOROOT} ${GOPATH}
+
+RUN [ ! -z ${GO_VERSION} ] && ( \
+      curl -L https://dl.google.com/go/go${GO_VERSION}.linux-amd64.tar.gz -o /tmp/go.tar.gz && \
+	  tar -xf /tmp/go.tar.gz -C ${GOROOT} --strip-components=1 && \
+      rm /tmp/go.tar.gz ) || \
+    echo "go is not required"
+
+ENV KONG_GO_PLUGINSERVER_INSTALL=${BUILD_PREFIX}/gps
+ENV KONG_GO_PLUGINSERVER=${KONG_GO_PLUGINSERVER}
+
+RUN [ ! -z ${KONG_GO_PLUGINSERVER} ] && ( \
+      go version && \
+	  mkdir ${KONG_GO_PLUGINSERVER_INSTALL} && \
+      cd ${KONG_GO_PLUGINSERVER_INSTALL} && \
+	  go mod init go-pluginserver && \
+	  go get -d -v github.com/Kong/go-pluginserver@${KONG_GO_PLUGINSERVER} && \
+	  go install -ldflags="-s -w -X main.version=${KONG_GO_PLUGINSERVER}" ... && \
+	  cd && \
+      rm -r ${KONG_GO_PLUGINSERVER_INSTALL} && \
+	  go-pluginserver --version ) || \
+    echo "Kong go pluginserver is not required"
```

Comment:
This block is not deterministic. gojira docker images are reused based on a name that should be deterministic https://github.com/Kong/gojira/blob/aa93e97433335b349d7c9dc7970806ef4ca8079a/gojira.sh#L524-L532. I know docker has sha layers for that, so we could just ignore the naming on images and go with whatever docker caches. Point here would be, based on two different `.requirements` file, we would get the same image name. We either add these conditionally too on the image name (which is already too long and would end up md5ing it).